src/utils/infinity.js

```diff
@@ -0,0 +1,387 @@
+         * The formatted time, such as "2:00 PM".
+         */
+        function convertHourMinute(hourMinute) {
+
+            var result = "",
+                time,
+                hour,
+                minute;
+
+            if (hourMinute && typeof hourMinute === "string" &&
+                hourMinute.length === 4 && hourMinute !== "    ") {
+
+                hour = parseInt(hourMinute.substring(0, 2));
+                minute = parseInt(hourMinute.substring(2, 4));
+
+                time = bbMoment({ hour: hour, minute: minute });
+                result = time.format("LT");
+
+            }
+
+            return result;
+
+        }
+
+        function toUpperIdOrNullIfEmpty(id) {
+            if (!id || id === GUID_EMPTY) {
+                return null;
+            }
+    
+            return id.toUpperCase();
+        }
+
+        // Using this function in order to keep from accidentally modifying
+        // an options object passed into a function.
+        function cloneOrNew(bbui, options) {
+            if (options) {
+                return bbui.clone(options);
+            }
+
+            return {};
+        }
+
+        return {
+            initialize: initialize,
+            getVirtualDirectory: getVirtualDirectory,
+            getDatabaseName: getDatabaseName,
+            getWebShellLoginUrl: getWebShellLoginUrl,
+            convertHourMinute: convertHourMinute,
+            toUpperIdOrNullIfEmpty: toUpperIdOrNullIfEmpty,
+            cloneOrNew: cloneOrNew,
+            configuration: {
+                isCustomApp: isCustomApp
+            }
+        };
+
+    }])
+
+    .factory("infinityAuth", ["infinityUtilities", "browserUtilities", "bbuiShellService", "$q",
+    function (infinityUtilities, browserUtilities, bbuiShellService, $q) {
+
+        var svc,
+            authenticateSuccessCallback,
+            authenticateFailureCallback,
+            authenticateFinallyCallback,
+            FORMS_AUTH_HEADER = "X-BB-FormsAuth",
+            noop = angular.noop;
+
+        function sessionStartSuccess(reply) {
+            authenticateSuccessCallback(reply.data);
+            authenticateFinallyCallback();
+        }
+
+        function sessionStartFailure(data, status, headers) {
+
+            var //formsAuthInUse,
+                redirectUrl;
+            //wsFederationEnabled,
+            //authHeader,
+            //isBearerAuthenticated,
+            //homePageUrl;
+
+            //homePageUrl = BBUI.urlConcat(svc.baseUrl, "browser/Default.aspx");
+            //homePageUrl += "?DatabaseName=" + encodeURIComponent(svc.databaseName);
+
+            //wsFederationEnabled = false;
+
+            //When using Federated Authentication, there may be a "Bearer" WWW-Authenticate header;
+            //But if the session start failed, all we can do is display the error on the login/migration form
+            //formsAuthInUse = (!authHeader || isBearerAuthenticated) || wsFederationEnabled;
```

Comment:
Should this be commented out? I think you should test with forms auth.

Reply:
This code was borrowed from other projects within our code base. I can remove the comments, or leave the code as is. Your call.

Reply:
This seems fine to me.